src/containers/SubjectViewer.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import SVGImage from '../components/SVGImage';
+import { Utility } from '../lib/Utility';
+
+import {
+  setRotation, setScaling, setTranslation,
+  setViewerState, SUBJECTVIEWER_STATE,
+} from '../ducks/subject-viewer';
+
+const INPUT_STATE = {
+  IDLE: 0,
+  ACTIVE: 1,
+}
+
+class SubjectViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    //HTML element refs.
+    this.section = undefined;
+    this.svg = undefined;
```

Comment:
Is it necessary to declare the refs as `undefined` initially? It seems a bit odd to me, and I was under the assumption that refs would be initiated once declared within the element.